pgjdbc/src/test/java/org/postgresql/test/jdbc2/ResultSetTest.java

```diff
@@ -1152,4 +1161,47 @@ private Map<String, Integer> getResultSetColumnNameIndexMap(ResultSet stmt) {
     return null;
   }
 
+  private static class CallableWithConnection implements Callable<Boolean> {
+
+    private final Connection connection;
+    private final int expectedYear;
+
+    protected CallableWithConnection(Connection connection, int expectedYear) {
+      this.connection = connection;
+      this.expectedYear = expectedYear;
+    }
+
+    @Override
+    public Boolean call() throws SQLException {
+      try (Statement statement = connection.createStatement()) {
+        for (int i = 0; i < 10; i++) {
+          try (ResultSet resultSet = statement.executeQuery(
+              String.format("SELECT unnest(array_fill('8/10/%d'::timestamp, ARRAY[%d]))",
+                  expectedYear, 500))) {
+            while (resultSet.next()) {
+              Timestamp d = resultSet.getTimestamp(1);
+              int year = 1900 + d.getYear();
+              if (year != expectedYear) {
+                return false;
+              }
```

Comment:
What if actual year is returned instead of true/false?

Reply:
done